test/e2e/framework/framework.go

```diff
@@ -105,17 +105,7 @@ func (f *Framework) BeforeEach() {
 		namespaceLabels := map[string]string{
 			"e2e-framework": f.BaseName,
 		}
-
-		for idx, clientSet := range f.ClusterClients {
-			switch idx {
-			case ClusterA: // On the first cluster we let k8s generate a name for the namespace
-				namespace := generateNamespace(clientSet, f.BaseName, namespaceLabels)
-				f.Namespace = namespace.GetName()
-				f.UniqueName = namespace.GetName()
-			default: // On the other clusters we use the same name to make tracing easier
-				f.CreateNamespace(clientSet, f.Namespace, namespaceLabels)
-			}
-		}
+		f.GenerateNamespace(f.BaseName, namespaceLabels)
```

Comment:
She wants to be able to call GenerateNamespace from other tests.

Reply:
* namespaceLabels generation should go inside GenerateNamespace, instead of being passed like a parameter.
* you don't need to pass basename, it's acessible already from "f" inside.


please do 
```
namespace := f.GenerateNamespace()
f.UniqueName = namespace
f.Namespace = namespace

```

Reply:
This lets us to provide custom labels rather than test generated labels. and also few of our NP tests would need different labeled namespaces.

Reply:
It's not exactly custom labels, but , extra generated labels. They're always generated.

Reply:
The idea here is namespacelabel for subm tests is done by 
namespaceLabels := map[string]string{
			"e2e-framework": f.BaseName,
		}

and for coastguard tests, it will be passed via f.GenerateNamespace(f.BaseName, namespaceLabels)